plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * @package plugins.elasticSearch
+ * @subpackage api.objects
+ */
+
+
+class KalturaESearchCategoryAggregationItem extends KalturaESearchAggregationItem
+{
+	/**
+	 *  @var KalturaESearchCategoryAggregateByFieldName
+	 */
+	public $fieldName;
+
+	public function toObject($object_to_fill = null, $props_to_skip = array())
+	{
+		if (!$object_to_fill)
+		{
+			$object_to_fill = new ESearchCategoryAggregationItem();
+		}
+		return parent::toObject($object_to_fill, $props_to_skip);
+	}
+
+	public function getFieldEnumMap()
+	{
+		return array (KalturaESearchCategoryAggregateByFieldName::CATEGORY_NAME => ESearchCategoryAggregationFieldName::CATEGORY_NAME);
+	}
+
+	protected function fixCategoryName($coreName)
+	{
+
+	}
+
+	public function coreToApiResponse($coreResponse, $fieldName = null)
+	{
+		$agg = new KalturaESearchAggregationResponseItem();
+		$agg->fieldName = $fieldName;
+		$agg->name = ESearchCategoryAggregationItem::KEY;
+		$bucketsArray = new KalturaESearchAggregationBucketsArray();
+		$buckets = $coreResponse[ESearchAggregations::BUCKETS];
+		if ($buckets)
+		{
+			foreach ($buckets as $bucket)
+			{
+				$responseBucket = new KalturaESearchAggregationBucket();
+				$responseBucket->fromArray($bucket);
+				list(,$categoryName) = elasticSearchUtils::reverseFormatCategoryNameStatus($responseBucket->value);
```

Comment:
what is the first parameter that is returned from the list? if not needed then why return 2 objects?

Reply:
Future use